src/tests/nativeaot/SmokeTests/Preinitialization/Preinitialization.cs

```diff
@@ -1674,6 +1675,82 @@ public static void Run()
     }
 }
 
+class TestVTables
+{
+    public static unsafe class IUnknownImpl
+    {
+        [FixedAddressValueType]
+        public static readonly IUnknownVftbl Vtbl;
+
+        static IUnknownImpl()
+        {
+            ComWrappers.GetIUnknownImpl(
+                fpQueryInterface: out *(nint*)&((IInspectableVftbl*)Unsafe.AsPointer(ref Vtbl))->QueryInterface,
+                fpAddRef: out *(nint*)&((IInspectableVftbl*)Unsafe.AsPointer(ref Vtbl))->AddRef,
+                fpRelease: out *(nint*)&((IInspectableVftbl*)Unsafe.AsPointer(ref Vtbl))->Release);
+        }
+    }
+
+    public static unsafe class IInspectableImpl
+    {
+        [FixedAddressValueType]
+        public static readonly IInspectableVftbl Vtbl;
+
+        static IInspectableImpl()
+        {
+            *(IUnknownVftbl*)Unsafe.AsPointer(ref Vtbl) = IUnknownImpl.Vtbl;
```

Comment:
Oh interesting, I thought the plan was not to support this. Is the advice for us to just do this in all our vtables then, rather than doing `ComWrappers.GetIUnknownImpl` for the `IUnknown` slots every time? Does this mean we can also do the same to copy the `IInspectable` vtable too as the first 6 slots, for CCW vtables that are inspectables?

Reply:
Yes, it looked allowable, hopefully Roslyn won't come up with too many ways to emit this in IL. The interpreter can be fragile, this has a happy path around `stobj` instruction.

Reply:
(Like I wrote in the issue, we can only make guarantees around IL sequences, not about C# source code. But since this is now a test in the repo, we'll get an advance notice if a break is incoming...)